framework/include/systems/NonlinearEigenSystem.h

```diff
@@ -114,7 +122,17 @@ class NonlinearEigenSystem : public NonlinearSystemBase
    * is the real and the imaginary part of
    * the eigenvalue, respectively.
    */
-  virtual const std::pair<Real, Real> getNthConvergedEigenvalue(dof_id_type n);
+  virtual const std::pair<Real, Real> getConvergedEigenvalue(dof_id_type n);
+
+  /**
+   * Return the Nth converged eigenvalue and copies the respective eigen vector to the solution
+   * vector.
+   *
+   * @return The Nth converged eigenvalue as a complex number, i.e. the first and the second number
+   * is the real and the imaginary part of
+   * the eigenvalue, respectively.
+   */
+  virtual const std::pair<Real, Real> getConvergedEigenpair(dof_id_type n);
```

Comment:
Same question here, although based on the doxygen, it looks like maybe you're writing some member data, so the method can't be marked `const`...?

Reply:
Will change